scheduler/plugin/wrappedplugin.go

```diff
@@ -0,0 +1,257 @@
+	// A Filter of the simulator plugin finally returns the status returned from AfterFilter.
+	AfterFilter(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo, filterResult *framework.Status) *framework.Status
+}
+
+type ScorePluginExtender interface {
+	// BeforeScore is a function that runs before the Score method of the original plugin.
+	// If BeforeScore returns non-success status, the simulator plugin doesn't run the Score method of the original plugin and return that score & status.
+	BeforeScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status)
+	// AfterScore is a function that runs after the Score method of the original plugin.
+	// A Score of the simulator plugin finally returns the score & status returned from AfterScore.
+	AfterScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string, score int64, scoreResult *framework.Status) (int64, *framework.Status)
+}
+
+type NormalizeScorePluginExtender interface {
+	// BeforeNormalizeScore is a function that runs before the NormalizeScore method of the original plugin.
+	// If BeforeNormalizeScore returns non-success status, the simulator plugin doesn't run the NormalizeScore method of the original plugin and return that status.
+	BeforeNormalizeScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status
+	// AfterNormalizeScore is a function that runs after the NormalizeScore method of the original plugin.
+	// A NormalizeScore of the simulator plugins finally returns the status returned from AfterNormalizeScore.
+	AfterNormalizeScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList, normalizeScoreResult *framework.Status) *framework.Status
+}
+
+// Extenders is a option for some pluginExtenders.
+// It will contain some arbitrary processing function defined by a user.
+type Extenders struct {
+	FilterPluginExtender         FilterPluginExtender
+	ScorePluginExtender          ScorePluginExtender
+	NormalizeScorePluginExtender NormalizeScorePluginExtender
+}
+
+type options struct {
+	extenderOption   Extenders
+	pluginNameOption string
+	weightOption     int32
+}
+
+type (
+	extendersOption  Extenders
+	pluginNameOption string
+	weightOption     int32
+)
+
+type Option interface {
+	apply(*options)
+}
+
+func (e extendersOption) apply(opts *options) {
+	opts.extenderOption = Extenders(e)
+}
+
+func (p pluginNameOption) apply(opts *options) {
+	opts.pluginNameOption = string(p)
+}
+
+func (w weightOption) apply(opts *options) {
+	opts.weightOption = int32(w)
+}
+
+func WithExtendersOption(opt *Extenders) Option {
+	return extendersOption(*opt)
+}
+
+func WithPluginNameOption(opt *string) Option {
+	return pluginNameOption(*opt)
+}
+
+func WithWeightOption(opt *int32) Option {
+	return weightOption(*opt)
+}
```

Comment:
Could you add brief explanations for each options?

Reply:
Thanks for the reviews! 
I added new comments for these options.